### ISSUE TITLE
Reached 'Managed Policies Per Role' Limit

### ISSUE THREAD

**Issue by blazinaj** [aws-amplify/amplify-cli#8116]
There is a hard limit on how many 'Managed Policies' can be attached to an IAM Role. The limit starts at 10 and can only be increased to 20.

Amplify CLI utilizes managed policies for the frontend 'auth' role, and adding new environment resources (API Models, Functions, etc), more policies must be generated and attached to the auth role.

We currently are at 19 policies and will soon hit the hard limit of 20, in our sandbox environments we have already been getting failed deployments when adding new resources.

What is our best recourse here? We use Amplify CLI in a production environment and have very active development going on.

**Comment by lazpavel** [comment]
Hello,

Thank you for raising the question. Currently Cognito can attach only one IAM role to an auth and unauth identity, which is a constraint and one role can have a limited set of policies. Can you provide the list of all managed policies tied to a single IAM role and then we can analyze it and see if we can optimize any of them?

**Comment by blazinaj** [comment]
What's the best way to get that to you privately? There are 19 managed policies that are full of account numbers and schema details

**Comment by lazpavel** [comment]
Hello @blazinaj,
Please send them over to amplify-cli@amazon.com
Thank you!

**Comment by blazinaj** [comment]
Sent them over, any workaround ideas would be greatly appreciated! This limit is putting us in a really bad spot, we've been developing using Amplify CLI for over 3 years now and are pretty heavily invested in the infrastructure here. Thank you for your prompt responses!

**Comment by blazinaj** [comment]
Yes. We have been unable to add new schema models for a long time because of this.

Stumbled upon [this other issue](https://github.com/aws-amplify/amplify-category-api/issues/275) that relates very closely. We've found that it doesn't even look like we need or want these policies to be made, since we are only using the `@iam` directive to allow our Lambdas to interact with the data on the backend, and these policies are being generated for the frontend. But we only use Cognito auth from the frontend. 

We can't figure out a way to get the CLI to not try and generate these policies. So every time we run a push operation, the entire pipeline fails when it attempts to generate the 21st policy

**Comment by josefaidt** [comment]
Hey @blazinaj apologies for the delay here, can you clarify how you've reached this limit? Is this after adding X amount of resources to an environment or via some other route?

**Comment by josefaidt** [comment]
Hey @blazinaj just wanted to follow up on this one and see if you are still experiencing this?

**Comment by blazinaj** [comment]
Hello, I apologize for the delay. 

Yes we are still experiencing this. This does come when we add the @iam directive to too many @model definitions. It adds too many managed policies to the authRole.

We have to use the @iam directive so that our lambdas can access the data, we are using purely Cognito on the front end. We would like to just disable this policy generation entirely, since we don't need most of them and this bricks our entire CI pipeline.
